src/views/open-api-docs-view/components/open-api-overview/components/status/utils/use-service-status.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { useEffect, useState } from 'react'
+import { ServiceStatus } from 'components/service-status-badge'
+
+/**
+ * Hook to use the StatusPage service status at a given endpoint.
+ */
+export function useServiceStatus(endpointUrl: string) {
+	const [status, setStatus] = useState<ServiceStatus>('loading')
+
+	useEffect(() => {
+		async function updateStatusData() {
+			setStatus(await fetchServiceStatus(endpointUrl))
+		}
+		updateStatusData()
```

Comment:
```suggestion
		(async function updateStatusData() {
			setStatus(await fetchServiceStatus(endpointUrl))
		})();
```
 ☁️  Really minor side-thought....Curious to hear your take on this versus the immediately invoked function. Style preference? Sometimes my brain thinks it could be easier to read the intent with the immediate function.

Reply:
Updated, but then I got some weird formatting, like:

```ts
useEffect(() => {
    ;(async function updateStatusData() {
        setStatus(await fetchServiceStatus(endpointUrl))
    })()
}, [endpointUrl])
```

Also tried an anonymous function for further terseness:

```ts
useEffect(() => {
    ;(async () => setStatus(await fetchServiceStatus(endpointUrl)))()
}, [endpointUrl])
```

To get rid of the start-of-line `;` (which I found a bit disorienting) while balancing clearly communicating the intent, I settled on:

```ts
useEffect(() => {
    const asyncEffect = async () => {
        setStatus(await fetchServiceStatus(endpointUrl))
    }
    asyncEffect()
}, [endpointUrl])
```

Let me know what you think! 🙇‍♂️